src/array.cr

```diff
@@ -1680,7 +1680,7 @@ class Array(T)
   # Raises `ArgumentError` if for any two elements the block returns `nil`.
   def unstable_sort(&block : T, T -> U) : Array(T) forall U
     {% unless U <= Int32? %}
-      {% raise "Expected block to return Int32 or Nil, not #{U}" %}
+      {% raise "Expected block to return Int32 or Nil, not #{U} (see Comparable#<=>)" %}
```

Comment:
I don't know if stdlib has a policy on tone for error messages or not, so please forgive the suggestion if I'm out of step. I had been planning to submit a pull like this, but admittedly wouldn't have been as comprehensive as you have been here. 

Looking at the [docs for Comparable#<=>](https://crystal-lang.org/api/1.12.2/Comparable.html#%3C%3D%3E%28other%3AT%29-instance-method), I'm not sure it would have actually helped me get what I wanted out of the situation. What do you think about an error message which is a little more prescriptive? Something like "did you mean to use #sort_by?" (or unstable_sort_by, sort_by!, etc).

Reply:
I'd agree we could be a bit more elaborate on this.
```suggestion
      {% raise "Expected block to return Int32 or Nil, not #{U}.\nThe block is supposed to be a custom comparison operation, compatible with `Comparable#<=>`.\nDid you mean to use `#sort_by?`" %}
```

Reply:
Does multiline compiler errors render well?

Reply:
Yeah, that should be fine.
```console
$ crystal eval '{% raise "Expected block to return Int32 or Nil.\nThe block is supposed to be a custom comparison operation, compatible with `Comparable#<=>`.\nDid you mean to use `#sort_by?`" %}'

Showing last frame. Use --error-trace for full trace.

error in line 1
Error: Expected block to return Int32 or Nil, not.

The block is supposed to be a custom comparison operation, compatible with `Comparable#<=>`.
Did you mean to use `#sort_by?`
```